### PR TITLE
8245956: JavaCompiler still uses File API instead of Path API in a specific case

### PR DESCRIPTION
**Pull request by lgxbslgx**
Hi all,

`JavacFileManager.getClassLoader` uses `getLocation` which causes exception in some situations.
This patch uses `getLocationAsPaths` instead of `getLocation` to solve it.
Thanks you for taking the time to review.

Best Regards.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8245956](https://bugs.openjdk.java.net/browse/JDK-8245956): JavaCompiler still uses File API instead of Path API in a specific case

### Reviewers
 * [Jonathan Gibbons](https://openjdk.java.net/census#jjg) (@jonathan-gibbons - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1553/head:pull/1553`
`$ git checkout pull/1553`

### CODE REVIEW

File: src/jdk.compiler/share/classes/com/sun/tools/javac/file/JavacFileManager.java [hunk 1]
```diff
@@ -741,13 +741,13 @@ public void close() throws IOException {
     @Override @DefinedBy(Api.COMPILER)
     public ClassLoader getClassLoader(Location location) {
         checkNotModuleOrientedLocation(location);
-        Iterable<? extends File> path = getLocation(location);
-        if (path == null)
+        Collection<? extends Path> paths = getLocationAsPaths(location);
```
Comment (jonathan-gibbons) [inline src/jdk.compiler/share/classes/com/sun/tools/javac/file/JavacFileManager.java:744]: Minor suggestion:  the use of the simple name `path` or `paths` can be confusing in contexts like this, when the name may refer to an instance of `java.nio.file.Path` or a collection of such items.

You did good by renaming the variable from `path` to `paths`.  A more explicit name would be to use `searchPath` as the name for a series of file system objects.

File: test/langtools/tools/javac/T8245956/T8245956.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8245956
+ * @summary JavaCompiler still uses files API instead of Path API in a specific case
```
Comment (jonathan-gibbons) [inline test/langtools/tools/javac/T8245956/T8245956.java:27]: I suggest changing `files API` to `File API`.
I have made the corresponding change in the JBS issue.

File: test/langtools/tools/javac/T8245956/T8245956.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        env.put("create", "true");
+        try (FileSystem fs = FileSystems.newFileSystem(jarZipFileUri, env)) {
+            Path fsPath = fs.getPath("");
+            JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
+            DiagnosticCollector<JavaFileObject> diagnosticCollector = new DiagnosticCollector<>();
+            try (StandardJavaFileManager fileManager = compiler.getStandardFileManager(diagnosticCollector, null, null)) {
+                List<Path> classPath = new ArrayList<>();
+                classPath.add(fsPath);
+                fileManager.setLocationFromPaths(StandardLocation.CLASS_PATH, classPath);
+                fileManager.getClassLoader(StandardLocation.CLASS_PATH);  // Should not generate any exceptions.
```
Comment (jonathan-gibbons) [inline test/langtools/tools/javac/T8245956/T8245956.java:62]: Minor suggestion: add something like a print statement after the `getClassLoader` call to verify that the `getClassLoader` call terminated normally.

**Review by jonathan-gibbons — APPROVED** [review]
Nice.

Some minor suggestions inline for your consideration.

**Review by jonathan-gibbons — APPROVED** [review]

**Comment by lgxbslgx** [comment]
@jonathan-gibbons I revise the code according to your suggestions. Thank you for taking the time to review.

**Review by jonathan-gibbons — APPROVED** [review]

**Pull request closed without merging**